client/app/app.vue

```diff
@@ -71,48 +72,45 @@
       },
 
       created () {
-          const query = this.$router.currentRoute.query
-
-          // Show Dialogs
-          if (query.dialogs) {
-              this.dialogs = query.dialogs.split(',').map((id) => {
-                  return Dialogs.find(item => item.id === id)
-              })
-          }
+          this.updateDialogs()
       },
 
       watch: {
-          dialogs (val, oldVal) {
-              let dialogs = val.map(item => item.id)
-
-              // Do not show dialogs query when empty
-              // avoid [].join(',') that leads to dialogs=''
-              if (dialogs.length) dialogs = dialogs.join(',')
-
-              // Update RouteQuery from dialogs values
-              const oldQuery = this.$router.currentRoute.query
-              const query = Object.assign({}, oldQuery, { dialogs })
-              this.$router.push({ query })
-          }
+          '$route': 'updateDialogs'
       },
 
       methods: {
-          onOpenDialog (id) {
-              const dialog = Dialogs.find(item => item.id === id)
-              if (-1 === this.dialogs.indexOf(dialog)) {
-                  this.dialogs.push(dialog)
+          updateDialogs () {
```

Comment:
I did a similar change here: https://github.com/cozy-labs/konnectors/pull/580/commits/292344a15b9bf45ce6b390dce4d31fc4030ebb28#diff-465b733d6d96531de99ee1a57bc74074R101

Anyway I hesitate to add this into methods or in a private function; I think your solution is better.